riscv64-embedded.hello cann't build

```bash
rewine@nixos /data [1]> nix build nixpkgs#pkgsCross.riscv64-embedded.hello
error: builder for '/nix/store/zhj0yn2brnzqwsy3dxj0m592xdb5xjqk-hello-riscv64-none-elf-2.12.drv' failed with exit code 2;
       last 10 log lines:
       > riscv64-none-elf-gcc -DLOCALEDIR=\"/nix/store/sf4j324myg29sgrydvxg9zhfjvxq45cb-hello-riscv64-none-elf-2.12/share/locale\" -DHAVE_CONFIG_H -I.  -Ilib -I./lib -Isrc -I./src   -g -O2 -c -o lib/getprogname.o lib/getprogname.c
       > lib/getprogname.c: In function 'getprogname':
       > lib/getprogname.c:287:4: error: #error "getprogname module not ported to this OS"
       >   287 | #  error "getprogname module not ported to this OS"
       >       |    ^~~~~
       > make[2]: *** [Makefile:2275: lib/getprogname.o] Error 1
       > make[2]: Leaving directory '/build/hello-2.12'
       > make[1]: *** [Makefile:2496: all-recursive] Error 1
       > make[1]: Leaving directory '/build/hello-2.12'
       > make: *** [Makefile:1899: all] Error 2
```

--- a/pkgs/applications/virtualization/spike/default.nix
+++ b/pkgs/applications/virtualization/spike/default.nix
@@ -1,26 +1,18 @@
-{ lib, stdenv, fetchFromGitHub, dtc, fetchpatch, pkgsCross }:
+{ lib, stdenv, fetchFromGitHub, dtc, pkgsCross }:
 
 stdenv.mkDerivation rec {
   pname = "spike";
-  version = "1.0.0";
+  version = "1.1.0";
 
   src = fetchFromGitHub {
     owner = "riscv";
     repo = "riscv-isa-sim";
     rev = "v${version}";
-    sha256 = "1hcl01nj96s3rkz4mrq747s5lkw81lgdjdimb8b1b9h8qnida7ww";
+    sha256 = "sha256-4D2Fezej0ioOOupw3kgMT5VLs+/jXQjwvek6v0AVMzI=";
   };
 
   nativeBuildInputs = [ dtc ];
   enableParallelBuilding = true;
-
-  patches = [
-    # Add missing headers to fix build.
-    (fetchpatch {
-      url = "https://github.com/riscv/riscv-isa-sim/commit/b3855682c2d744c613d2ffd6b53e3f021ecea4f3.patch";
-      sha256 = "1v1mpp4iddf5n4h3kmj65g075m7xc31bxww7gldnmgl607ma7cnl";
-    })
-  ];
 
   postPatch = ''
     patchShebangs scripts/*.sh
@@ -39,7 +31,9 @@ stdenv.mkDerivation rec {
     ''
       runHook preInstallCheck
 
-      $out/bin/spike -m64 ${riscvPkgs.riscv-pk}/bin/pk ${riscvPkgs.hello}/bin/hello | grep -Fq "Hello, world"
+      echo -e "#include<stdio.h>\nint main() {printf(\"Hello, world\");return 0;}" > hello.c
+      ${riscvPkgs.stdenv.cc}/bin/riscv64-none-elf-gcc -o hello hello.c
+      $out/bin/spike -m64 ${riscvPkgs.riscv-pk}/bin/pk hello | grep -Fq "Hello, world"
 
       runHook postInstallCheck
     '';

